test/WebSites/ValidationWebSite/Controllers/ModelMetadataTypeAttributeController.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using ValidationWebSite.ViewModels;
+
+namespace ValidationWebSite.Controllers
+{
+    public class ModelMetadataTypeValidationController : Controller
+    {
+        [HttpPost]
+        public object ValidateProductViewModelInclMetadata([FromBody] ProductViewModel product)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object ValidateSoftwareViewModelInclMetadata([FromBody] SoftwareViewModel software)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object TryValidateModelProductViewModelWithErrorInParameter(int id, [FromBody] ProductViewModel product)
+        {
+            if (id == 0)
+            {
+                ModelState["id"].Errors.Clear();
+            }
+            TryValidateModel(product);
```

Comment:
this action and the associated test remain unclear.  what error would the `ModelState["id"].Errors.Clear()` remove?  why is the removal conditional -- no test sends anything but `0`?

I suggested the `[Required, MaxLength(3), StringLength(10) { MinLength = 5 }] string theImpossibleString` approach to make sending a completely valid request impossible.  the action should then start with `ModelState["theImpossibleString"].Errors.Clear();` and otherwise ignore the parameter.